PR - Code review - Behaviour of mobile menu

wickathou opened the pull request:
For this PR, a mobile menu has been implemented

Features:
- Added HTML content
- CSS rules for mobile display of the menu
- JS to add event listeners to buttons

wickathou committed:
Added JS to control behaviour of mobile menu
Added custom css rules for mobile behavior

wickathou committed:
Fixed lint issues

wickathou committed:
Added readme content

Johnadibe commented on README.md:
```diff
@@ -1,2 +1,76 @@
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://wickathou.github.io/)
```
> * [x] I noticed that the live demo link in your Readme file does not match the design of the developed website. Please update the live demo link to reflect the current design of the website or remove it. ☺️.

Johnadibe commented on index.html:
```diff
@@ -0,0 +1,461 @@
+      </button>
+      <nav id="mobile-menu" class="not-active">
+        <div>
+          <i id="close" class="fas fa-xmark"></i>
+          <ul>
+            <li><a class="mobile-lin" href="#projects">Portoflio</a></li>
+            <li><a class="mobile-link" href="#about-m">About</a></li>
+            <li>
+              <a class="mobile-link" href="#contact">Contact</a>
+            </li>
+            <i id="close" class="fas fa-xmark"></i>
+          </ul>
```
> * [x] There are two i tags that appear on the mobile screen when the hamburger menu is open. To provide the best user experience for your website visitors, it is important to ensure that there are no duplicates or unnecessary elements that could confuse or distract them. Kindly update your website's code to remove one of the close buttons on the mobile screen to match the required design.😊
> 
> ![hamburger](https://user-images.githubusercontent.com/40009354/233188390-356e55f0-4a9f-4473-b19c-9689f56061e5.PNG)

Johnadibe reviewed (changes requested):
# Status: Required Changes ♻️
## Hi @wickathou👋, 

### I'm John, your PR reviewer

You have made good progress so far! However, there are still some issues that require your attention before moving on to the next project. 

You are very close to reaching your goal, so keep up the good work! 🚀

> ![](https://user-images.githubusercontent.com/20563252/121369265-4e418880-c901-11eb-848c-1eff06d5a4cf.gif)
> 

## Highlights
- Well-structured files :heavy_check_mark: 
- UI matched required design :heavy_check_mark: 
- Good use of JavaScript, HTML&CSS best practices :heavy_check_mark: 
- Good job for using the correct GitHub flow:heavy_check_mark: 
- All linters are passing :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 


## Required Changes ♻️
> * [x]  **I noticed that the mobile menu on your portfolio website does not disappear when a user clicks on the "portfolio" navigation link. According to the requirements, the mobile menu should disappear when a user clicks or taps any of the options. This ensures that the user has a seamless experience while navigating your website. Kindly fix that.😊**
> 
> > * _When the user clicks (or taps) any of the mobile menu options, the mobile menu disappears._
>

> * [x] **There is an issue with the "about" navigation link not displaying the correct section of the page when clicked. Please rectify this issue. Fixing this issue will improve the user experience😊**
> > * _When the user clicks (or taps) any of the mobile menu options, the correct part of the page is displayed._
>

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

wickathou committed:
Fixed
Deleted live view from readme
Fixed about links
Fixed portfolio link not closing navbar

wickathou committed:
Modified Readme file

wickathou committed:
Fixed

jmonto55 commented on README.md:
```diff
@@ -1,2 +1,84 @@
+Thanks Microverse, learn how to code > [Join Microverse](https://www.microverse.org/?grsf=9m3hq6)
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
+
```
[OPTIONAL]  Great job tagging your MIT License inside your README file 👌. However, please kindly make sure that the link is working in order to make your repo look more professional🤓, below is what I get when I click on the link 👇🏽.

![license](https://user-images.githubusercontent.com/116701485/233440335-de738c01-429d-490e-8e7d-2b85a38ddb66.png)

jmonto55 commented on index.html:
```diff
@@ -0,0 +1,460 @@
+      <nav id="mobile-menu" class="not-active">
+        <div>
+          <i id="close" class="fas fa-xmark"></i>
+          <ul>
+            <li><a class="mobile-link" href="#projects">Protfoliu</a></li>
```
[OPTIONAL] Good work building your `nav` list with semantic tags 🤓.  However, please kindly make sure to check if the word 'Portfolio' is spelled right, even if this typo doesn't affect the functionality of your code it can be interpreted as a lack of attention to detail.

jmonto55 commented on index.html:
```diff
@@ -0,0 +1,460 @@
+          <i id="close" class="fas fa-xmark"></i>
+          <ul>
+            <li><a class="mobile-link" href="#projects">Protfoliu</a></li>
+            <li><a class="mobile-link" href="#aAbout-me">About</a></li>
+            <li>
+              <a class="mobile-link" href="#c#ontact">Contact</a>
```
- [ ] Great job building your `nav` list with semantic tags 💯. However, please kindly make sure that when the user clicks on any of the options from the menu the correct section of the page is displayed as requested by the previous reviewer. Check the requirement below 👇🏽.
  >When the user clicks (or taps) any of the mobile menu options, the correct part of the page is displayed.
  
Currently, when I click on About or Contact, the mobile menu is closed but the correct section of the page is not being shown 👇🏽

https://user-images.githubusercontent.com/116701485/233445725-beea7f78-78bd-495c-a28a-4f92db7bf1ed.mp4

jmonto55 commented on css/main.css:
```diff
@@ -0,0 +1,609 @@
+  color: #fff5e1;
+}
+
+#mobile-menu li {
+  border-bottom: #fff5e1 solid 1px;
+  margin-top: 5%;
+  text-align: start;
+}
```
[OPTIONAL ] You did good work styling your mobile menu `li` items ✔️. However, please kindly make sure to stick as much as possible to the Figma design, including font size, font weight, and the same icons as requested in the requirement below 👇🏽.
  >You must stick to the design as much as possible (e.g., font, colors, images, text, margins) using [the templates in Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1).
  
 This is the Figma design 👇🏽

![figma](https://user-images.githubusercontent.com/116701485/233448244-3da062eb-fb30-4929-8ed3-19228c2f43b7.png)


This is you design 👇🏽 (almost there 🤓)

![design](https://user-images.githubusercontent.com/116701485/233448966-f6a8b00c-9ce8-4b62-a30a-9f77e0d6bfaa.png)

jmonto55 reviewed (changes requested):
Hi @wickathou 👋🏽, 

Great work on making the changes requested by a previous reviewer 👌
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![AlmostThereGIF](https://user-images.githubusercontent.com/116701485/232347631-60851d91-4b6b-4c48-aa78-13e4bc4dc267.gif)
 ## Highlights

- [ ] Great use of HTML semantic tags ✅

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
